packages/blockchain-extension/extension/webview/ReactTutorialGalleryView.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { ReactView } from './ReactView';
+import { ExtensionUtil } from '../util/ExtensionUtil';
+import { Reporter } from '../../extension/util/Reporter';
+import * as fs from 'fs-extra';
+
+export class ReactTutorialGalleryView extends ReactView {
+    constructor(context: vscode.ExtensionContext) {
+        super(context, 'tutorialGallery', 'Tutorial Gallery');
+    }
+
+    async openPanelInner(panel: vscode.WebviewPanel): Promise<void> {
+        Reporter.instance().sendTelemetryEvent('openedView', {openedView: panel.title}); // Report that a user has opened a new panel
+
+        const extensionPath: string = ExtensionUtil.getExtensionPath();
+        const panelIcon: vscode.Uri = vscode.Uri.file(path.join(extensionPath, 'resources', 'logo.svg'));
+
+        panel.iconPath = panelIcon;
+
+        /* will need something like this to open the tutorials from the react app */
+        // panel.webview.onDidReceiveMessage(async (message: {command: string, data?: any}) => {
+        //     if (message.data) {
+        //         await vscode.commands.executeCommand(message.command, ...message.data);
+        //     } else {
+        //         await vscode.commands.executeCommand(message.command);
+        //     }
+        // });
+
+        await this.loadComponent(panel);
+    }
+
+    async readTutorialFiles(): Promise<Array<{name: string, contents: string[]}>> {
+        const tutorialData: Array<{name: string, contents: string[]}> = [];
+
+        const extensionPath: string = ExtensionUtil.getExtensionPath();
+        const tutorialsPath: string = path.join(extensionPath, 'tutorials');
+        const tutorialDirectories: string[] = await fs.readdir(tutorialsPath);
+
+        const otherTutorials: {name: string, contents: string[]} = {
+            name: 'Other tutorials',
+            contents: []
+        };
+        for (const directory of tutorialDirectories) {
+            const directoryPath: string = path.join(extensionPath, 'tutorials', directory);
+            const directoryContents: string[] = await fs.readdir(directoryPath);
+            if (directory === 'Basic tutorials') { // hardcoding this for now until we have more tutorial series from the engagement team
```

Comment:
Do we want to replace this with something like `tutorials.json`, and logic which makes it easy to add new tutorials and series?